### PR TITLE
Unit tests

### LINKED ISSUE #3: make unit test, give an example

### PR DESCRIPTION
**Pull request by volodymyrss**

### CODE REVIEW

**Comment by volodymyrss** [comment]
this will need https://github.com/oda-hub/dispatcher-app/pull/69

**Comment by volodymyrss** [comment]
Note that the one more "real" test successfully fails at the moment. I did not attempt understading what a meaningful request would be. But I was amused by the sort of error message it returns, maybe it points to another bug, maybe even in dispatcher.

We also discussed with @burnout87  that similar approach can be adopted for other plugins

**Comment by burnout87** [comment]
I think this is the approach to follow, also for the other plugins

File: README.md [hunk 1]
```diff
@@ -10,18 +10,18 @@ ANTARES cdci plugin is distributed under the terms of The MIT License.
...
+Denys Savchenko, Andrea Tramacere
 
 ISDC Data Centre for Astrophysics, Astronomy Department of the University of Geneva, Chemin d'Ecogia 16, CH-1290 Versoix, Switzerland
 
 Configuration for deployment
 ----------------------------
-- copy the conf_file from 'cdci_antares_plugin/config_dir/data_server_conf.yml' and place in given directory
-- set the env var `CDCI_ANTARES_PLUGIN_CONF_FILE` to the path of the file conf_file 
-- edit the in conf_file the two keys:
+- copy the `conf_file` from `cdci_antares_plugin/config_dir/data_server_conf.yml' and place in given directory
```
Comment (burnout87) [inline README.md]: Minor: perhaps also 'config'  will have to be inserted in the vocab list

File: README.md [hunk 2]
```diff
@@ -10,18 +10,18 @@ ANTARES cdci plugin is distributed under the terms of The MIT License.
...
+- set the environment variable `CDCI_ANTARES_PLUGIN_CONF_FILE` to the path of the file conf_file 
+- edit the in `conf_file` the two keys:
```
Comment (burnout87) [inline README.md]: I might got confused, what are the two keys to edit?

**Review by burnout87 — CHANGES_REQUESTED** [review]

**Comment by dsavchenko** [comment]
> Note that the one more "real" test successfully fails at the moment.

Of course, as to process real query it relies on backend. 

> But I was amused by the sort of error message it returns, maybe it points to another bug, maybe even in dispatcher.

Not in the dispatcher. I even put a comment and then forgot about it https://github.com/oda-hub/dispatcher-plugin-antares/blob/fcb60bbd6ebe9669f8862e308d970b80394d1e50/dispatcher_plugin_antares/antares_dataserver_dispatcher.py#L217

**Comment by volodymyrss** [comment]
> > Note that the one more "real" test successfully fails at the moment.
> 
> Of course, as to process real query it relies on backend.

Alright! Could be there is some meaningful dummy-like query?
Or you could install it as a requirement and test it with a another fixture. If it is possible.

If not, the test should be really considered an "integration", in sense that it will be run on a specific instance, deployed for review.
I.e. it will not work with github actions.
In this case, the backend access will be configured on defining the test.
You can test it while setting your own instance.
And we test it after deploying in unige and also regularly.

> 
> > But I was amused by the sort of error message it returns, maybe it points to another bug, maybe even in dispatcher.
> 
> Not in the dispatcher. I even put a comment and then forgot about it
> 
> https://github.com/oda-hub/dispatcher-plugin-antares/blob/fcb60bbd6ebe9669f8862e308d970b80394d1e50/dispatcher_plugin_antares/antares_dataserver_dispatcher.py#L217

ok! It's really a kind of feature common in the dispatcher through. So see as you

**Comment by dsavchenko** [comment]
Ideologically it will look more like an integration test

without backend it will fail unless query_type='Dummy' (which itself is the desirable behaviour, which could be tested in units)

For query_type='Dummy' it should inherit the proper dummy behaviour from the base classes defined in dispatcher, no need to implement something specific. Am I right?

File: README.md [hunk 3]
```diff
@@ -10,19 +10,14 @@ ANTARES cdci plugin is distributed under the terms of The MIT License.
 
 Who's responsible?
 -------------------
-Andrea Tramacere
+Denys Savchenko, Andrea Tramacere
```
Comment (dsavchenko) [inline README.md:14]: @andriineronov should I put here Paris? APC or IPGP or something?

**Comment by andriineronov** [comment]
Just skip "CDCI"

File: README.md [hunk 4]
```diff
@@ -10,19 +10,14 @@ ANTARES cdci plugin is distributed under the terms of The MIT License.
...
 
 ISDC Data Centre for Astrophysics, Astronomy Department of the University of Geneva, Chemin d'Ecogia 16, CH-1290 Versoix, Switzerland
```
Comment (dsavchenko) [inline README.md:16]: @andriineronov I am about this line

Comment (ferrigno) [reply README.md:16]: Remove "ISDC Data Centre for Astrophysics," it does not exist

**Review by burnout87 — APPROVED** [review]

**Comment by dsavchenko** [comment]
> For query_type='Dummy' it should inherit the proper dummy behaviour from the base classes defined in the dispatcher, no need to implement something specific. Am I right?

I am not fully right. I got: `get_dummy_products needs to be implemented in derived class`. 

So let's stick with what is here as a baseline. Then I will implement this method and use Dummy query in test

File: README.md (hunk 4 above)
Comment (volodymyrss) [reply README.md:16]: we could really also do these changes in a separate PR, it's not very clear if we are ok with the changes suggested here.

**Review by dsavchenko — APPROVED** [review]

**Comment by volodymyrss** [comment]
> > For query_type='Dummy' it should inherit the proper dummy behaviour from the base classes defined in the dispatcher, no need to implement something specific. Am I right?
> 
> I am not fully right. I got: `get_dummy_products needs to be implemented in derived class`.
> 
> So let's stick with what is here as a baseline. Then I will implement this method and use Dummy query in test

you are very right though that it's good to stick with the baseline and separate other details features to elsewhere!

**Pull request merged**